### PR TITLE
Add TBD Readme

### LINKED ISSUE #5515: Add basic README for e2e testing
#### Summary

Lately we have added a lot of capabilities in the area of e2e testing. We should document these at a basic level and provide a place where we can track e2e testing information, best-practices, etc for the team and open-source community at large.

### PR DESCRIPTION
**Pull request by unlikelyzero**
Closes #5515 

### Describe your changes:

Adds a README covering the basics of what/why/how etc in regards to e2e testing on Open MCT.

### All Submissions:

* [x] Have you followed the guidelines in our [Contributing document](https://github.com/nasa/openmct/blob/master/CONTRIBUTING.md)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/nasa/openmct/pulls) for the same update/change?
* [x] Is this change backwards compatible? For example, developers won't need to change how they are calling the API or how they've extended core plugins such as Tables or Plots.

### Author Checklist

* [x] Changes address original issue?
* [ ] Unit tests included and/or updated with changes?
* [x] Command line build passes?
* [x] Has this been smoke tested?
* [ ] Testing instructions included in associated issue OR is this a dependency/testcase change?

### Reviewer Checklist

* [ ] Changes appear to address issue?
* [ ] Changes appear not to be breaking changes?
* [ ] Appropriate unit tests included?
* [ ] Code style and in-line documentation are appropriate?
* [ ] Commit messages meet standards?
* [ ] Has associated issue been labelled unverified? (only applicable if this PR closes the issue)
* [ ] Has associated issue been labelled bug? (only applicable if this PR is for a bug fix)

### CODE REVIEW

File: e2e/README.md [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+## Overview
+
+e2e testing, sometimes known as "end to end" testing is an industry standard approach to automating the testing of Web UIs like Open MCT. Broadly speaking, e2e tests differentiate themselves in that they err on the side of replicating real user interaction over executing raw javascript functions like unit tests. 
+
+Historically, the abstraction necessary to replicate real user behavior meant:
+- that e2e frameworks were not capable of validing codepaths which were only reachable by unit tests
+- were expensive to run due to the need to instantiate a real browser
+- were flaky due to network conditions or the underlying protocols associated with testing a browser. 
+
+However, as the web ecosystem has matured to the point where Mission Critical UIs are written for the Web (Open MCT), the e2e testing tools improved along with them. There are now fewer "trade-offs" when chosing to write an e2e test over any other type of test. Modern e2e frameworks:
```
Comment (akhenry) [inline e2e/README.md:14]: Typo, "choosing"

File: e2e/README.md [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+- enable browsers and web applications to reach 'mocked' states to cover deeper codepaths without mocking the browser, itself
```
Comment (akhenry) [inline e2e/README.md:15]: No need for a comma here.

File: e2e/README.md [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+- have direct access to javascript component libraries like vue.js props
+- are now more reliable than the application-under-test due to raw and native access to browser-internal protocols. 
+- These new browser-internal protocols enable near-instant bi-directional communication between test code browser speeding up test execution and making the tests as reliable as the application, itself
+
+Furthermore, the abstraction necessary to run e2e tests as a user, better enables them to be extended to run in a variety of contexts. This matches the extensible design of Open MCT. A single e2e test in Open MCT is extended to run:
+- Against a matrix of browser versions
+- Against a matrix of OS platforms
+- Against a local development version of openmct
```
Comment (akhenry) [inline e2e/README.md:23]: Open MCT is two words

File: e2e/README.md [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+
+This document captures information specific to e2e testing of Open MCT. For general information about testing, please see [testing section](link to testing section).
+
+## Overview
+
+e2e testing, sometimes known as "end to end" testing is an industry standard approach to automating the testing of Web UIs like Open MCT. Broadly speaking, e2e tests differentiate themselves in that they err on the side of replicating real user interaction over executing raw javascript functions like unit tests. 
+
+Historically, the abstraction necessary to replicate real user behavior meant:
+- that e2e frameworks were not capable of validing codepaths which were only reachable by unit tests
+- were expensive to run due to the need to instantiate a real browser
```
Comment (jvigliotta) [inline e2e/README.md:11]: if you're flowing in from "meant:" I'd recommend adding "e2e tests were expensive" in the beginning here.

File: e2e/README.md [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+- were flaky due to network conditions or the underlying protocols associated with testing a browser.
```
Comment (jvigliotta) [inline e2e/README.md:12]: same as above, recommend adding "e2e tests were flaky" at the beginning.

**Review by akhenry — CHANGES_REQUESTED** [review]

File: e2e/README.md (hunk 1 above)
Comment (jvigliotta) [inline e2e/README.md:14]: I'd say, ", the e2e testing tools have matured as well."

Comment (jvigliotta) [reply e2e/README.md:14]: Chosing -> Choosing

**Review by ozyx — APPROVED** [review]

**Pull request merged**